### PR TITLE
[#2001] Extract c-zoom-commit-message component from views/c-zoom

### PR DESCRIPTION
**Pull request by sopa301**
Part of #2001.

### Proposed commit message

```
Many of the frontend files are difficult to navigate through due to the
large file sizes. As we add more features to the frontend, it is
getting harder to maintain. This is also very unfriendly towards new
contributors. 

Let us break down frontend files in a logical manner, continuing with
extracting c-zoom-commit-message from views/c-zoom.
```

### CODE REVIEW

File: frontend/src/components/c-zoom-commit-message.vue [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+      required: true,
+    },
+    info: {
+      type: Object as PropType<{ zUser: User | undefined, zAvgContributionSize: number, zIsMerged: Boolean }>,
+      required: true,
+    },
+    day: {
+      type: Object as PropType<Commit>,
+      required: true,
+    },
```
[the remark this answers is not in the document]
Comment (sopa301) [reply frontend/src/components/c-zoom-commit-message.vue]: Nope! Thanks for catching it!

**Review by ckcherry23 — APPROVED** [review]
LGTM! Thanks for helping with the componentization.

**Review by MarcusTXK — CHANGES_REQUESTED** [review]
Looks great, nice job with this, it makes `c-zoom.vue` alot easier to read. I have just a minor nitpick with the spacing of the `methods`, as some have an extra new lines after them and others do not.

Shall we standardize it as no spaces between the methods?

**Comment by sopa301** [comment]
Okay, should we add a [linting rule](https://eslint.style/rules/ts/lines-between-class-members) for that?

**Comment by MarcusTXK** [comment]
> Okay, should we add a [linting rule](https://eslint.style/rules/js/padding-line-between-statements) for that?

Yup sure, feel free to create an issue for this, I think that will definitely help with consistency

**Review by MarcusTXK — APPROVED** [review]
LGTM

**Pull request merged**